### PR TITLE
IndexTrigger AOP

### PR DESCRIPTION
**Pull request by ducquoc**
initial implementation, unit test included

### CODE REVIEW

**Comment by ducquoc** [comment]
It seems I misunderstood about the design of the index trigger. This implementation intercepts all Cassandra Thrift insertion and if there is a systemProperty "solrHost" (URL of Solr server) it will send an HTTP request to Solr URL to update/add the index.
Now that I think again, if the design of the all triggers are to register particular triggers to events of certain column family (and with the enable/pause functionality), then the implementation does not have to intercept all insertions or writing CommitLog (i.e. not necessary AOP). 
In other words, the IndexTrigger just sends the HttpRequest and leaves the event binding/registration for TriggerStore/TriggerTask . If the target URL can handle the updating index too it would be even more simple.
